codalab/rest/bundles.py

```diff
@@ -1310,6 +1314,16 @@ def delete_bundles(uuids, force, recursive, data_only, dry_run):
             # Just remove references to the data hashes
             local.model.remove_data_hash_references(relevant_uuids)
         else:
+            # If the bundle is stored on cloud, first delete data on cloud.
+            for uuid in relevant_uuids:
+                bundle_location = local.bundle_store.get_bundle_location(uuid)
+
+                file_location = '/'.join(bundle_location.split('/')[0:-1]) + "/"
+                if bundle_location.startswith(
+                    StorageURLScheme.AZURE_BLOB_STORAGE.value
+                ) or bundle_location.startswith(StorageURLScheme.GCS_STORAGE.value):
+                    FileSystems.delete([file_location])
```

Comment:
Just want to make sure this works with both GCS and Azure, since I know in the upload case they had to be dealt with differently

Reply:
> Just want to make sure this works with both GCS and Azure, since I know in the upload case they had to be dealt with differently


Yes it works for both Azure and GCS. Because this part of code is running on Codalab server, while bypass upload happens at the client side. At user client side, we need handle Azure and GCS differently.